### ISSUE TITLE
Security: "Only the current user may connect to the network" MUST be default, not the other way around.

### ISSUE THREAD

**Issue by Lukas-UAUX** [eduvpn/python-eduvpn-client#504]
As I found out today you implemented and released a new feature lately to allow installing the VPN as a system wide profile. 

Although there is quite the issue there with certificate based authentication an having the certificate placed in a users home for a system wide config (It is not FHS compliant and would fail if root can not access these files ege. with NFS homes...) having the option itself is fine. (This is NetworkManagers Problem to solve...)

**BUT: NEVER EVER, should this be turned 'off' by default. This is a extreme security risk especially on shared devices. If just anyone could use the connection with my certificates an impersonate me _without even knowing._** 

The user *must* know what he is doing with this option and has to turn it on manually to share the connection.

**Comment by Lukas-UAUX** [comment]
(Also we do not allow credential sharing, this option undermines it automatically and makes users do forbidden stuff without even knowing)

**Comment by jwijenbergh** [comment]
> As I found out today you implemented and released a new feature lately to allow installing the VPN as a system wide profile.

No, we added the ability to have the profile for the current user only. Thus eliminating the sudo popup you get on some systems (depends on your policykit settings). It is by default in networkmanager that profiles are system wide, so before this all of our profiles were system wide.

I did not enable this by default because this is also not the way it works in networkmanager. This is off by default in networkmanager. You can check this by opening nm-connection-editor and looking at the box "all users may connect to this network", this is enabled by default.

> BUT: NEVER EVER, should this (Only the current user may connect to the network) be turned 'off' by default. This is a extreme security risk especially on shared devices. If just anyone could use the connection with my certificates an impersonate me without even knowing.

Please take a look at your policykit settings https://wiki.archlinux.org/title/NetworkManager#Set_up_PolicyKit_permissions.

I can enable it by default as I see where your concern is coming from. I was following the practice of other VPNs as they also do not seem to do this. I see that you already made an issue in the networkmanager page https://gitlab.gnome.org/GNOME/NetworkManager-openvpn/-/issues/89

**Comment by Lukas-UAUX** [comment]
> No, we added the ability to have the profile for the current user only.

It was already default before the change / implementation...  That was how I found out about the issue in NM which lead to the creation of https://gitlab.gnome.org/GNOME/NetworkManager-openvpn/-/issues/89 (which nobody seems to care about...)

Installing through the eduVPN-Client resulted in "current user only" while a manual import using NM was importing it system-wide per default. 

> You can check this by opening nm-connection-editor and looking at the box "all users may connect to this network", this is enabled by default.

With a password-based VPNs this is mostly fine as one has to enter credentials again when switching users. The Problem is only there, when personal certificates are used - because NM just stupidly keeps the user-specific paths for them as system-config (and no keychain is used) and as it is running as root itself it allows access to them them which leads to dangerous impersonations, and it mostly invalidates authentication.

As the Client is mainly targeted at devices from end users (students) trying to get a user to configure polkit or else to something that is not already default by distro is a hopeless effort if it is not strictly necessary for using a service. (see eduroam and configuring ca-certificates...)

> I can enable it by default as I see where your concern is coming from.
That would be very nice!

**Comment by jwijenbergh** [comment]
> It was already default before the change / implementation... That was how I found out about the issue in NM which lead to the creation of https://gitlab.gnome.org/GNOME/NetworkManager-openvpn/-/issues/89 (which nobody seems to care about...). Installing through the eduVPN-Client resulted in "current user only" while a manual import using NM was importing it system-wide per default.

I am sorry but this is not possible unless you changed something in your system settings. See https://github.com/eduvpn/python-eduvpn-client/pull/489/files, the code path for enabling user only connections was not present before. Reverting to an earlier release will give you system wide connections by default. See also https://github.com/eduvpn/python-eduvpn-client/issues/488

3.0.0:
![image](https://user-images.githubusercontent.com/46386452/187244262-02c8dac0-1fe7-48f3-9c46-236b5c1740f3.png)

Is it maybe possible that you changed this setting? The eduVPN client saves the uuid of the connection and modifies it for new servers

> With a password-based VPNs this is mostly fine as one has to enter credentials again when switching users. The Problem is only there, when personal certificates are used - because NM just stupidly keeps the user-specific paths for them as system-config (and no keychain is used) and as it is running as root itself it allows access to them them which leads to dangerous impersonations, and it mostly invalidates authentication.

I am afraid that the bulk of this issue needs to be fixed in networkmanager ;)

> As the Client is mainly targeted at devices from end users (students) trying to get a user to configure polkit or else to something that is not already default by distro is a hopeless effort if it is not strictly necessary for using a service. (see eduroam and configuring ca-certificates...)

The use case you're describing of shared devices is very rare on linux, especially for students. My advice is that shared devices  have strict policykit settings anyways. But I will enable it by default unless users will report that there are problems.

Note that I am currently doing a huge refactor of the app by making it simpler and more secure. With 'more secure' I am also planning to implement keychain integration such that the eduVPN config is encrypted, thus not revealing your tokens if someone/a program reads your config. This does mean that a new release might take a while. Unless you think this is important enough to warrant a new hotfix release.

**Comment by Lukas-UAUX** [comment]
> I am sorry but this is not possible unless you changed something in your system settings.

Funny, I actually checked it on a fresh install before posting the issue to the gnome gitlab... Maybe only the Certificate was ommited? Moving on..

> Note that I am currently doing a huge refactor of the app by making it simpler and more secure. With 'more secure' I am also planning to implement keychain integration such that the eduVPN config is encrypted, thus not revealing your tokens if someone/a program reads your config. 

That sounds nice, I add a warning in our knowledge base for our end users for now (as I did for stock NM). And I notify our internal users...

**Comment by jwijenbergh** [comment]
Pr made in #507, w ill be merged soon so users can grab it from master.
